### PR TITLE
chore(examples): migrate SSR example to angular 7

### LINKED ISSUE #379: SSR Question / Error: Angular 7 + docs example results in AlgoliaSearchUnknownError
Describe the bug 🐛

2 things: 

**1) When following the example here: https://community.algolia.com/angular-instantsearch/guides/server-side-rendering.html I get the following server side error:**

```  name: 'AlgoliaSearchUnknownError',
  message: 'Unknown error occured',
  debugData:
   [ { currentHost: 'https://<removed>-dsn.algolia.net',
       headers: [Object],
       content:
        '{"requests":[{"indexName":"<removed>","params":"query=&page=0&highlightPreTag=__ais-highlight__&highlightPostTag=__%2Fais-highlight__&facets=%5B%5D&tagFilters="}]}',
       contentLength: 177,
       method: 'POST',
       timeouts: [Object],
       url: '/1/indexes/*/queries',
       startTime: 2018-11-18T02:14:56.414Z,
       endTime: 2018-11-18T02:14:56.415Z,
       duration: 1 } ] }
```
trace
```
ERROR { AlgoliaSearchUnknownError: Unknown error occured
    at tryFallback (/Users/cwd/dist/server.js:232925:15)
    at ZoneDelegate.invoke (/Users/cwd/dist/server.js:531:26)
    at Object.onInvoke (/Users/cwd/dist/server.js:18006:33)
    at ZoneDelegate.invoke (/Users/cwd/dist/server.js:530:32)
    at Zone.run (/Users/cwd/dist/server.js:281:43)
    at /Users/cwd/dist/server.js:1015:34
    at ZoneDelegate.invokeTask (/Users/cwd/dist/server.js:564:31)
    at Object.onInvokeTask (/Users/cwd/dist/server.js:17997:33)
    at ZoneDelegate.invokeTask (/Users/cwd/dist/server.js:563:36)
    at Zone.runTask (/Users/cwd/dist/server.js:331:47)
```
server.ts
```
app.engine('html', (_, options, callback) => {
  const engine = ngExpressEngine({
    bootstrap: AppServerModuleNgFactory,
    providers: [
      { provide: 'request', useFactory: () => options.req, deps: [] },
      provideModuleMap(LAZY_MODULE_MAP)
    ]
  });
  engine(_, options, callback);
});
```
component config (in my constructor)
```
this.config = {
                searchParameters,
                urlSync: false,
                indexName: `SecuredUser_${environment.name}`,
                searchClient:  createSSRSearchClient({
                    makeStateKey,
                    HttpHeaders,
                    transferState: this.transferState,
                    httpClient: this.httpClient,
                    appId: '<removed>',
                    apiKey: '<removed>'
                })
            };
```

**2) What is the point of createSSRSearchClient vs the standard config? Specificially, is there a simple reason why SSR doesn't "just work" with a standard implementation of instant search?** 

To Reproduce 🔍
check out this commit (which still contains webpack): https://github.com/angular/universal-starter/commit/d85b67038a18b0a2cc20e7eb8e1b9964b413ebea

update deps in package.json to latest angular versions

follow the docs:  https://community.algolia.com/angular-instantsearch/guides/server-side-rendering.html

### PR DESCRIPTION
**Pull request by tkrugg**
**Summary**
We used to have a hard time getting SSR to work ever since Angular 6 came out.
I've updated the project to `@angular/*` and `@nguniversal` ^7. It seems better.

**Result**

I'll have to do the same thing again with Angular 6 just to check if we can completely remove any compatibility caveat from our docs, but I won't be making a PR for that

closes #379

### CODE REVIEW

File: examples/server-side-rendering/angular.json [hunk 1]
```diff
@@ -0,0 +1,141 @@
+{
```
Comment (tkrugg) [inline examples/server-side-rendering/angular.json]: This is the new `.angular-cli.json` introduced by the migration to `@angular/cli@^7`

File: examples/server-side-rendering/src/polyfills.ts [hunk 2]
```diff
@@ -39,7 +39,6 @@
 
 /** Evergreen browsers require these. **/
 import 'core-js/es6/reflect';
-import 'core-js/es7/reflect';
```
Comment (tkrugg) [inline examples/server-side-rendering/src/polyfills.ts]: This was removed to the migration.
It's is expected when [upgrading to v7](https://blog.angular.io/version-7-of-angular-cli-prompts-virtual-scroll-drag-and-drop-and-more-c594e22e7b8c#f14f)

File: src/version.ts [hunk 3]
```diff
@@ -1 +1 @@
-export const VERSION = "2.2.1";
+export const VERSION = '2.2.1';
```
Comment (Haroenv) [inline src/version.ts]: this change seems unrelated

File: examples/server-side-rendering/tsconfig.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
   "compileOnSave": false,
   "compilerOptions": {
+    "importHelpers": true,
```
Comment (samouss) [inline examples/server-side-rendering/tsconfig.json]: This is a new requirement?

**Comment by samouss** [comment]
Rather than "WIP xxx" use the draft pull request!

screenshot 2019-02-20 at 12 41 47

File: examples/server-side-rendering/tsconfig.json (hunk 4 above)
Comment (tkrugg) [reply examples/server-side-rendering/tsconfig.json]: This one is another result of the migration.
In package.json you'll also notice a new dependency [tslib].(https://www.npmjs.com/package/tslib) introduced by the migrator (I think it's optional)
This compile option is a usage requirement of this lib.

Comment (samouss) [reply examples/server-side-rendering/tsconfig.json]: Yep it look like it's the same purpose than Babel runtime.

File: examples/server-side-rendering/server.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import 'zone.js/dist/zone-node';
 import 'reflect-metadata';
-import { renderModuleFactory } from '@angular/platform-server';
```
Comment (tkrugg) [inline examples/server-side-rendering/server.ts]: This was unused

File: examples/server-side-rendering/webpack.server.config.js [hunk 6]
```diff
@@ -4,25 +4,24 @@ const path = require('path');
 const webpack = require('webpack');
 
 module.exports = {
+  mode: 'development',
```
Comment (tkrugg) [inline examples/server-side-rendering/webpack.server.config.js]: This is a workaround for this issue: https://github.com/angular/angular-cli/issues/8616

Comment (francoischalifour) [reply examples/server-side-rendering/webpack.server.config.js]: It'd be nice to add this explanation in the code itself, so that next time we come across this line we know why it's here.

Comment (tkrugg) [reply examples/server-side-rendering/webpack.server.config.js]: yes will do!

File: src/version.ts (hunk 3 above)
Comment (tkrugg) [reply src/version.ts]: it's just a prettier doing it's thing
